ansible/roles/kernel/tasks/os/debian.yml

```diff
@@ -12,27 +12,30 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 - block:
   - name: Reboot
     reboot:
-
   - name: Gather facts after reboot
     setup:
-
   when: reboot
 
-- name: Install {{ansible_os_family}} Family Kernel Packages
-  apt:
+- name: Install current and future kernel headers
+  ansible.builtin.apt:
     name:
     - linux-headers-{{ansible_kernel}}
-    state: latest
-    update_cache: yes
-  when: reboot
-
-- name: Install {{ansible_os_family}} Family Kernel Packages
-  apt:
-    name:
-    - linux-headers-generic
-    state: latest
-    update_cache: yes
-  when: not reboot
+    - "{{ debian_family_kernel_headers_package | default('linux-headers-cloud-amd64') }}"
```

Comment:
Do I understand that we install the future headers in the case that the user manually upgrades the kernel, allowing for nvidia drivers to be recompiled without needed a public network connection?

Reply:
The headers are necessary for any kernel module to compile. NVIDIA is one.

`linux-headers-cloud-amd64` is the metapackage by which upgrades of the "real" package are managed. So you can install the real package (`linux-headers-{{ansible_kernel}}`) but without the metapackage, you won't get upgrades when you also upgrade the kernel. So this is what I mean by "future" kernels.

Debian installations have the metapackage for the kernel installed by default but not the metapackage for the headers.